src/coreclr/jit/gentree.cpp

```diff
@@ -25150,6 +25175,40 @@ bool GenTreeHWIntrinsic::OperIsMemoryStoreOrBarrier() const
     }
 }
 
+//------------------------------------------------------------------------
+// OperIsBroadcastScalar: Is this HWIntrinsic a broadcast node from scalar.
+//
+// Return Value:
+//    Whether "this" is a broadcast node from scalar.
+//
+bool GenTreeHWIntrinsic::OperIsBroadcastScalar() const
+{
+#if defined(TARGET_XARCH)
+    NamedIntrinsic intrinsicId = GetHWIntrinsicId();
+    switch (intrinsicId)
+    {
+        case NI_AVX2_BroadcastScalarToVector128:
+        case NI_AVX2_BroadcastScalarToVector256:
+            return true;
+        default:
+            return false;
+    }
+#else
+    return false;
+#endif
+}
+
+//------------------------------------------------------------------------
+// OperIsEmbBroadcastHWIntrinsic: Checks if the intrinsic is a embedded broadcast compatible inintrsic.
+//
+// Return Value:
+// true if the intrisic node lowering instruction is embedded broadcast compatible.
+//
+bool GenTreeHWIntrinsic::OperIsEmbBroadcastHWIntrinsic() const
```

Comment:
I'd suggest renaming this to just `OperIsEmbBroadcastCompatible`.

That helps clarify we're checking for `embedded broadcast compatibility` and doesn't reiterate that its a `HWIntrinsic` (since its already only accessible from `GenTreeHWIntrinsic` nodes)